README.md

```diff
@@ -450,6 +454,65 @@ stdio to an in-memory [virtual terminal](https://github.com/hinshun/vt10x).
 
 For some examples, you can see any of the tests in this repo.
 
+### Using a mock
+
+Instead of calling the survey functions directly, you can create a survey struct and call the functions from there.
+
+```golang
+
+survey := survey.Survey{}
+
+response := false
+prompt := &survey.Confirm{
+    Message: "Do you like pie?",
+}
+survey.AskOne(prompt, &response)
+
+```
+
+If you create only one survey struct at the top level of your program and pass it to all functions, you can test those functions by replacing the struct with a mock provided by survey.
+
+#### main
+
+```golang
+func main() {
+    survey := survey.Survey{}
+
+    AskForPie(survey)
+}
+
+func AskForPie(survey survey.SurveyInterface) bool {
+    response := false
+    prompt := &survey.Confirm{
+        Message: "Do you like pie?",
+    }
+    survey.AskOne(prompt, &response)
+
+    return response
+}
+
+```
+
+#### Test
+
+```golang
+func TestAskForPie(t *testing.T) {
+    
+    //create mock
+    mock := survey.SurveyMock{} 
+    //set the response the "user" should select 
+    mock.setResponse(true)
+
+    result := AskForPie(mock)
+    
+    //check output of the function
+    if result != true {
```

Comment:
```suggestion
    if !result {
```

Reply:
I explicitly added this comparison for clarity for less experienced developers, but the short form is probably understandable enough as well. 

I'll leave the decision up you.

Reply:
I think it does more of a service to inexperienced developers to show them code that reflects the idiomatic style for that language. I appreciate that you tried to add clarity to documentation examples, but I do not think that logical negation operator will pose a barrier to understanding Survey.